saleor/order/utils.py

```diff
@@ -59,6 +59,10 @@ def recalculate_order(order, **kwargs):
     Voucher discount amount is recalculated by default. To avoid this, pass
     update_voucher_discount argument set to False.
     """
+    # do not use prefetched order lines, cause they might have changed
+    if hasattr(order, '_prefetched_objects_cache'):
```

Comment:
This is quite uncommon, wouldn't `order.refresh_from_db()` do the trick (if it fetches related objects which I'm not sure of)? Anyway, it's very intriguing that we have to remove prefetched data.

Reply:
Unhopefully, `refresh_from_db` doesn't work. Don't have any other ideas right now.